src/common/droid-util.c

```diff
@@ -2498,15 +2620,37 @@ size_t pa_droid_stream_buffer_size(pa_droid_stream *s) {
     return s->buffer_size;
 }
 
-bool pa_sink_is_droid_sink(pa_sink *s) {
-    const char *api;
+void pa_droid_stream_set_data(pa_droid_stream *s, void *data) {
+    pa_assert(s);
+
+    s->data = data;
```

Comment:
if s->data is already set, does something need to be done about it before assigning new value?